extensions/kogito/deployment/src/main/java/io/quarkus/kogito/deployment/KogitoAssetsProcessor.java

```diff
@@ -0,0 +1,224 @@
+            boolean generateRuleUnits, boolean generateProcesses) throws IOException {
+        Path targetClassesPath = root.getPath();
+        Path projectPath = targetClassesPath.toString().endsWith("target/classes") ? targetClassesPath.getParent().getParent()
+                : targetClassesPath;
+
+        String appPackageName = "org.kie.kogito.app";
+
+        ApplicationGenerator appGen = new ApplicationGenerator(appPackageName, new File(projectPath.toFile(), "target"))
+                .withDependencyInjection(true);
+
+        if (generateRuleUnits) {
+            appGen.withGenerator(RuleCodegen.ofPath(projectPath, launchMode == LaunchMode.DEVELOPMENT))
+                    .withRuleEventListenersConfig(customRuleEventListenerConfigExists(projectPath, appPackageName));
+        }
+
+        if (generateProcesses) {
+            appGen.withGenerator(ProcessCodegen.ofPath(projectPath))
+                    .withWorkItemHandlerConfig(
+                            customWorkItemConfigExists(projectPath, appPackageName))
+                    .withProcessEventListenerConfig(
+                            customProcessListenerConfigExists(projectPath, appPackageName));
+        }
+
+        return appGen;
+    }
+
+    private String customWorkItemConfigExists(Path projectPath, String appPackageName) {
+        String sourceDir = Paths.get(projectPath.toString(), "src").toString();
+        String workItemHandlerConfigClass = ProcessCodegen.defaultWorkItemHandlerConfigClass(appPackageName);
+        Path p = Paths.get(sourceDir,
+                "main/java",
+                workItemHandlerConfigClass.replace('.', '/') + ".java");
+        return Files.exists(p) ? workItemHandlerConfigClass : null;
+    }
+
+    private String customProcessListenerConfigExists(Path projectPath, String appPackageName) {
+        String sourceDir = Paths.get(projectPath.toString(), "src").toString();
+        String processEventListenerClass = ProcessCodegen.defaultProcessListenerConfigClass(appPackageName);
+        Path p = Paths.get(sourceDir,
+                "main/java",
+                processEventListenerClass.replace('.', '/') + ".java");
+        return Files.exists(p) ? processEventListenerClass : null;
+    }
+
+    private String customRuleEventListenerConfigExists(Path projectPath, String appPackageName) {
+        String sourceDir = Paths.get(projectPath.toString(), "src").toString();
+        String ruleEventListenerConfiglass = RuleCodegen.defaultRuleEventListenerConfigClass(appPackageName);
+        Path p = Paths.get(sourceDir,
+                "main/java",
+                ruleEventListenerConfiglass.replace('.', '/') + ".java");
+        return Files.exists(p) ? ruleEventListenerConfiglass : null;
+    }
+
+    private String toRuntimeSource(String className) {
+        return "src/main/java/" + className.replace('.', '/') + ".java";
```

Comment:
good point, will take a look at possibilities to have this replaced

Reply:
just was going through docs of Gradle (not really a Gradle user) and it seems that they follow the same folder structure so src/main/java seems to be safe. I agree that if someone configured the source directory differently then it will break but don't think we can make it anyway better without taking this from the maven/gradle plugin somehow. Or am I missing something obvious?

Reply:
I agree, right now we don't pass the source dir to the extension. So, that's good for now.